src/Opdex.Platform.Application/EntryHandlers/Addresses/Mining/GetMiningPositionsWithFilterQueryHandler.cs

```diff
@@ -29,10 +32,16 @@ public override async Task<MiningPositionsDto> Handle(GetMiningPositionsWithFilt
 
         var positionsResults = positions.ToList();
 
+        _logger.LogTrace("Retrieved queried mining positions");
+
         var cursorDto = BuildCursorDto(positionsResults, request.Cursor, pointerSelector: result => result.Id);
 
+        _logger.LogTrace("Returning {ResultCount} results", positionsResults.Count);
+
```

Comment:
First thoughts, this is going to be quite a bit of logs, not sure what level we need to log at but it could be quite a bit.

Might be more efficient to wrap handlers within a timer and write logs with "starting" and "finished in ##ms" rather than step by step logs.

Reply:
A timer doesn't really add anything as time is included for all logs. It's good to be generous with trace logs, as they're the lowest granular level. We only ever need to turn them on when we want to trace execution time or debug something very granular. The idea behind putting in these logs is that we can see when database results get returned, when the cursor is built and when assembling the dtos happens. This will hopefully give us an idea which of these 3 steps might be the cause of performance concerns. Only logging at the start and end of handler execution won't provide this sort of insight.

Reply:
I hear ya but...this will be **a lot** of logs and quite a bit of effort to go back and implement everywhere. 

It might be beneficial to take a simple route first, then come back and get granular if/where necessary.

We could create a new BaseHandler, with protected `StartTrace` and `StopTrace` methods and easily call those base methods passing in the name of the parent handler at the start and end of `Handle()`. Then we get the base info we need for a very minimal amount of work.

``` C#
// Imperfect example
public override async Task<MiningPositionsDto> Handle(GetMiningPositionsWithFilterQuery request, CancellationToken cancellationToken) 
{
    StartTrace();
    
    // Our actual handler logic
    
    StopTrace();
}

// Base Class Methods
protected void StartTrace() {
    _logger.LogTrace("{0}: Starting handler execution", _handlerName);
    _timer = new Stopwatch();
    _timer.Start();
}

protected void StopTrace() 
{
    _timer.Stop();
    _logger.LogTrace("{0}: Time elapsed: {1:hh\\:mm\\:ss}", _handlerName, stopwatch.Elapsed);
}
```

Reply:
I don't disagree with logging around handlers, it'll produce a lot of useful info.

This level of granularity is necessary due to the result of the load test. I'm not sure whether by *a lot* you're suggesting it's too granular? We won't have trace logs turned on unless we need them, really I don't see the downside of having a lot of logs when all of them have a purpose. 

```csharp
...
// gives us an idea of how long the database query takes
_logger.LogTrace("Retrieved queried address balances");

 var cursorDto = BuildCursorDto(balancesResults, request.Cursor, pointerSelector: result => result.Id);

 // tells us how long building the cursor takes as well as how many dtos will be assembled
 _logger.LogTrace("Returning {ResultCount} results", balancesResults.Count);

 var dtos = await Task.WhenAll(balancesResults.Select(balance => _assembler.Assemble(balance)));

 // lets us know how long assembling of the dtos takes
 _logger.LogTrace("Assembled results");
```

In fact, maybe removing the `Retrieved queried .........` logs and logging more granularly within the select handlers or `DbContext` instead would be more beneficial, so we can see a more precise result for query time.

Reply:
It's a code smell having this many logs around and it's a bit tedious to continue doing. 

We know a total requests time and we can see the full flow of a request through the traceId. Logging these only in the infrastructure layer is probably better, maybe the Cirrus Modules before the go out to the Node, CMC modules going to their API, and DB context.

If we do this, theres only a few places that need this logging and we could add a DB context optional param at the end for the `operationName` so we'll know exactly which query it was. We'd just add `nameof(OurQueryHandler)`.

it might be also worth using a StopWatch and logging the elapsed time within these logs rather than us looking at DateTimes within logs. 

None of our internal operations should be long running I don't think without needing the help of this external information.

Reply:
I disagree about it being a code smell, but I'll remove the logs within these handlers and happy to replace with logging around the external calls like you suggest. That should allow us to gather a good amount of information to work with.